File: 242-valid-anagram/242-valid-anagram.java

class Solution {
    public boolean isAnagram(String s, String t) {
        
        int n1 =s.length();
        int n2 =t.length();
        
        int[] count1 = new int[26];
        int[] count2 = new int[26];
        
        if(n1!=n2)
            return false;
        
        for(int i=0;i<n1;i++){
            count1[s.charAt(i)-'a']++;
        }
        
        for(int j=0;j<n2;j++){
            count2[t.charAt(j)-'a']++;
        }
        
        for(int i=0;i<26;i++){
            if(count1[i]!=count2[i])
               return false; 
        }
        return true;
    }
}

File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
       int n1=s.size();
       int n2=t.size();
       if(n1!=n2)
           return false;
        
        unordered_map<char,int>map;
        
        for(int i=0;i<n1;i++){
            map[s[i]]++;
        }
        
        for(auto i:t){
            map[i]--;
            if(map[i]<0)
                return false;
        }
        return true;
    }
};